jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -232,6 +232,11 @@ def _parse_hdulist(app, hdulist, file_name=None,
             metadata[PRIHDR_KEY] = standardize_metadata(hdulist['PRIMARY'].header)
 
         sc = _return_spectrum_with_correct_units(flux, wcs, metadata, data_type, hdulist=hdulist)
+
+        # store original WCS in metadata. this is a hacky workaround for converting subsets
+        # to sky regions, where the parent data of the subset might have dropped spatial WCS info
+        metadata['_orig_spatial_wcs'] = wcs.celestial
```

Comment:
I don't know how I feel about introducing yet one more hidden meta key to hack around Cubeviz WCS stuff. Why not use the existing "_orig_spec"? Does doing that introduce too big a performance penalty? This comment applies to all other similar occurrences.

```suggestion
        metadata['_orig_spec'] = sc
```

Reply:
we only sometimes need the full sc, but (for now) we always need the wcs (just the celestial component too) so i decided to always store that since its inexpensive. it will also be easier to refactor (rather than putting a check in if sc is not stored, then store wcs, then unpacking it later and doing the same check for sc)